plantseg/pipeline/utils.py

```diff
@@ -48,3 +50,62 @@ def __init__(self, log_queue):
 
     def emit(self, record):
         self.log_queue.put(record)
+
+
+def find_input_key(h5_file):
+    # if only one dataset in h5_file return it, otherwise return first from H5_KEYS
+    found_keys = list(h5_file.keys())
+    if not found_keys:
+        raise RuntimeError(f"No datasets found in '{h5_file.filename}'")
+
+    if len(found_keys) == 1:
+        return found_keys[0]
+    else:
+        for h5_key in H5_KEYS:
+            if h5_key in found_keys:
+                return h5_key
+
+        raise RuntimeError(f"Ambiguous datasets '{found_keys}' in {h5_file.filename}")
+
+
+def read_tiff_voxel_size(file_path):
+    """
+    Implemented based on information found in https://pypi.org/project/tifffile
+    """
+
+    def _xy_voxel_size(tags, key):
+        assert key in ['XResolution', 'YResolution']
+        if key in tags:
+            num_pixels, units = tags[key].value
+            return units / num_pixels
+        # return default
+        return 1.
+
+    with tifffile.TiffFile(file_path) as tiff:
+        image_metadata = tiff.imagej_metadata
+        if image_metadata is not None:
+            z = image_metadata.get('spacing', 1.)
+        else:
+            # default voxel size
+            z = 1.
+
+        tags = tiff.pages[0].tags
+        # parse X, Y resolution
+        y = _xy_voxel_size(tags, 'YResolution')
+        x = _xy_voxel_size(tags, 'XResolution')
+        # return voxel size
+        return [z, y, x]
+
+
+def read_h5_voxel_size(file_path):
```

Comment:
I this is intended to be used in `GenericPipelineStep.load_stack`, but is currently unused.

Reply:
Yes and no. It is used in the gui for automatic scaling. I did not want to touch the GenericPipelineStep in this PR. But I can add it